freeipa/src/main/java/com/sequenceiq/freeipa/kerberosmgmt/v1/AuthorizationUtil.java

```diff
@@ -0,0 +1,34 @@
+package com.sequenceiq.freeipa.kerberosmgmt.v1;
+
+import com.sequenceiq.cloudbreak.auth.altus.Crn;
+import com.sequenceiq.cloudbreak.auth.altus.GrpcUmsClient;
+import com.sequenceiq.cloudbreak.logger.MDCUtils;
+import org.springframework.security.access.AccessDeniedException;
+import org.springframework.stereotype.Component;
+
+import javax.inject.Inject;
+
+@Component
+public class AuthorizationUtil {
+
+    @Inject
+    private GrpcUmsClient grpcUmsClient;
+
+    /**
+     * An authorization check that verifies that a calling actor can perform an operation against a
+     * target user. The calling actor only requires the right when the target user is different.
+     */
+    public void checkCallerIsSelfOrHasRight(String actorCrnStr, String targetUserCrnStr, String requiredRight) {
+        Crn actorCrn = Crn.safeFromString(actorCrnStr);
```

Comment:
assert the parameters?

Reply:
The safeFromString does a null check for us.